wweval/R/plotting_style.R

```diff
@@ -0,0 +1,115 @@
+#' If they are dates, we will rotate x axis tick text 45 degrees. Default is
+#' `FALSE`
+#'
+#' @return a theme object to add to a [ggplot2::ggplot()] object to specify
+#' line size and formatting
+#' @export
+get_plot_theme <- function(x_axis_text_size = 8,
+                           y_axis_text_size = 8,
+                           x_axis_title_size = 12,
+                           y_axis_title_size = 12,
+                           legend_title_size = 10,
+                           legend_text_size = 8,
+                           facet_x_text_size = 8,
+                           facet_y_text_size = 8,
+                           plot_title_size = 10,
+                           x_axis_dates = FALSE) {
+  ww_theme <-
+    cowplot::theme_half_open() +
+    cowplot::background_grid() +
+    theme(
+      axis.text.x = element_text(
+        size = x_axis_text_size
+      ),
+      axis.text.y = element_text(
+        size = y_axis_text_size
+      ),
+      legend.title = element_text(size = legend_title_size),
+      legend.text = element_text(size = legend_text_size),
+      strip.text.x = element_text(size = facet_x_text_size),
+      strip.text.y = element_text(size = facet_y_text_size),
+      axis.title.x = element_text(size = x_axis_title_size),
+      axis.title.y = element_text(size = y_axis_title_size),
+      plot.title = element_text(
+        size = plot_title_size,
+        vjust = 0.5, hjust = 0.5
+      )
+    )
+
+
+  if (isTRUE(x_axis_dates)) {
+    # If x-axis are dates, default to 2 week date breaks
+    # and rotate 45 degrees
+    ww_theme <- ww_theme +
+      theme(
+        axis.text.x = element_text(
+          size = x_axis_text_size - 2, vjust = 1,
+          hjust = 1, angle = 45
+        )
+      )
+  }
+
+  return(ww_theme)
+}
+
+
+#' Get plot components (colors for now)
+#'
+#' @return a list of model colors and horizon colors to be passed to
+#' `scale_fill_manual()` and `scale_color_manual()`
+#' @export
+plot_components <- function() {
+  pal_horizons <- RColorBrewer::brewer.pal(12, "Paired")
+
+  # somewhat arbitrary, we can play with these later
+  horizon_colors <- c(
+    "calibration" = pal_horizons[1],
+    "nowcast" = pal_horizons[2],
+    "1 wk" = pal_horizons[3],
+    "2 wks" = pal_horizons[4],
+    "3 wks" = pal_horizons[5],
+    "4 wks" = pal_horizons[6],
+    "overall" = pal_horizons[10]
+  )
+
+  pal_model <- RColorBrewer::brewer.pal(8, "Dark2")
+
+  model_colors <- c(
+    "ww" = pal_model[1],
+    "hosp" = pal_model[2],
+    "cfa-wwrenewal(retro)" = pal_model[1],
+    "cfa-wwrenewal(real-time)" = pal_model[5],
+    "cfa-hosponlyrenewal(retro)" = pal_model[2],
+    "COVIDhub-4_week_ensemble" = pal_model[3],
+    "UMass-trends_ensemble" = pal_model[4],
+    "UT-Osiris" = pal_model[6],
+    "COVIDhub-baseline" = pal_model[7]
+  )
+
+  colors_list <- list(
+    horizon_colors = horizon_colors,
+    model_colors = model_colors
+  )
+  return(colors_list)
+}
```

Comment:
I'd actually create a `scale_color_model()`, `scale_color_horizon()` etc from your pre-computed `horizon_colors`, e.g.

```R
scale_color_horizon <- function(...) {
   return (scale_colour_manual(values = unlist(horizon_colors), ...))
}
```

Then you can do:

```R
my_plot <- my_plot + scale_color_horizon() + other_ggplot_stuff()
```

Reply:
I would still need to pass in the `colors` though right? So this would just avoid using `values = colors$horizon_colors` and instead allow you to do
`scale_color_horizon(colors)`?

Reply:
no, in this approach the `scale_fill_` function would "know" the colors. Concrete example from some paper code of mine where I defined a color scale for variants:

First I defined the variant-name-to-color mapping as a `list`:

https://github.com/dylanhmorris/variant-aerosol-stability/blob/d23b9cd9b3030915ab950b1ed1de2725805a6529/varstab/R/plotting_style.R#L221-L236


Then I used that `list` to create a custom `scale_color_variant()` function:

https://github.com/dylanhmorris/variant-aerosol-stability/blob/d23b9cd9b3030915ab950b1ed1de2725805a6529/varstab/R/plotting_style.R#L256-L270

Reply:
If I'd also needed the particular values of that list to be accessible in other project code via a function call, I could have defined another function, e.g. something like this:

````R
get_variant_color <- function(variant_name) {
    if(!(variant_name %in% names(variant_colors)) {
        cli::cli_abort("No color defined for variant {variant_name}")
   }
   return (variant_colors[[variant_name]])
}

Reply:
@dylanhmorris Those links aren't working for me for some reason. 

So I made those lists in `plot_components()` but I think it just isn't clear to me how to make that accessible in other project code without simply calling the function (which is what I have in here).

Reply:
Oops. Try the links now. Should explain a bit more what I mean. In this first, the list mapping values to colors exists at script scope in the file where the functions that use it are defined (but it is not exported).

Reply:
Ok, I see -- so in this example you are just creating the mapping of variants to colors, then writing a function that uses that mapping by default, rather than including the mapping within the function itself.

If you'd prefer, I can rewrite the `plot_components()` to do this for the things that I have set to have specific colors (which are models, horizons, and phases).

Reply:
Let me know if I am understanding the ask correctly!